src/physical_object.py

```diff
@@ -33,16 +33,39 @@ def get_forces_composition(self):
             force[1] += self.forces[f][1]
         return force
 
+    def collide(self, q):
+        """
+        Dirtily calculates collisions.
+
+        :q: other object
+
+        :returns: new speed
+        """
+        s = self
```

Comment:
As long as it is clear what it takes, shouldn't matter. Although if I used p and q to represent different things in different places, that would make perfect sense.